src/utilities/routeResolver.ts

```diff
@@ -1,17 +1,19 @@
-import { Resolved, Routes, isParentRoute, isNamedRoute, Route } from '@/types'
+import { Resolved, Routes, isParentRoute, isNamedRoute, Route, Param } from '@/types'
 import { mergeParams, path as createPath } from '@/utilities'
 
-export function resolveRoutes(routes: Routes, parentPath = '', parentParams = {}): Resolved<Route>[] {
+type RouteParentContext = {
+  parentPath: string,
+  parentParams: Record<string, Param[]>,
+  parentNames: string[],
+}
```

Comment:
this is how I chose to get around the max args count of 3

Reply:
I like this pattern anyway where we have one required argument and then a second optional argument that has additional contextual stuff in it 👍